code/modules/surgery/organs/organ.dm

```diff
@@ -57,7 +57,7 @@ var/list/organ_cache = list()
 /obj/item/organ/proc/set_dna(var/datum/dna/new_dna)
 	if(new_dna)
 		dna = new_dna.Clone()
-		blood_DNA.Cut()
+		blood_DNA = list()
```

Comment:
generally speaking, when you're nulling out a list, it's best to deliberately use `Cut` (despite being a hair TINY bit more expensive). But yes, memory leaks.